output/clickhouse/clickhouse.go

```
@@ -0,0 +1,120 @@
+}
+
+// DefaultOutputConfig returns an OutputConfig struct with default values
+func DefaultOutputConfig() OutputConfig {
+	return OutputConfig{
+		OutputConfig: config.OutputConfig{
+			CommonConfig: config.CommonConfig{
+				Type: ModuleName,
+			},
+		},
+		DSN:               "http://localhost:8123",
+		BulkSize:          1000,
+		BulkFlushInterval: 5 * time.Second,
+	}
+}
+
+// InitHandler initialize the output plugin
+func InitHandler(ctx context.Context, raw *config.ConfigRaw) (config.TypeOutputConfig, error) {
+	conf := DefaultOutputConfig()
+	err := config.ReflectConfig(raw, &conf)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create http client.
+	// @todo: allow user to set these parameters
+	conf.httpClient = pester.New()
+	conf.httpClient.Concurrency = 1
+	conf.httpClient.MaxRetries = 0
+	conf.httpClient.Backoff = pester.ExponentialBackoff
+	conf.httpClient.KeepLog = true
+
+	return &conf, nil
+}
+
+// Output event
+// @todo: create bulk functionality
+// @todo: add gzip functionality
+func (t *OutputConfig) Output(ctx context.Context, event logevent.LogEvent) (err error) {
+
+	newEventMap := make(map[string]interface{})
+
+	for _, key := range t.Fields {
+		fieldValue, exists := event.GetValue(key)
+
+		if true == exists {
+			switch fieldValue.(type) {
+			case time.Time:
+				timeValue, err := time.Parse(`2006-01-02 15:04:05.999999999 -0700 MST`, fmt.Sprintf("%s", fieldValue))
+				if err != nil {
+					log.Printf("Error parsing time field %s", err)
+				}
+				newEventMap[key] = timeValue.Format("2006-01-02 15:04:05")
+			default:
+				newEventMap[key] = fieldValue
+			}
+		}
+
+	}
+
+	newEventAsJSON, err := jsoniter.Marshal(newEventMap)
+
+	if err != nil {
+		return err
+	}
+
+	queryString := fmt.Sprintf("INSERT INTO %s FORMAT JSONEachRow", t.TargetTable)
+	requestURL := fmt.Sprintf("%s?query=%s", t.DSN, url.QueryEscape(queryString))
+
+	request, _ := http.NewRequest("POST", requestURL, strings.NewReader(string(newEventAsJSON)))
```

Please do not ignore any errors.

---

Sure thing, will do.